### DISCUSSION TITLE
Group license ERROR this regfile is not for this device

### DISCUSSION THREAD

**Discussion by parseltongued** [dashingsoft/pyarmor#1293] [Q&A]
Hi

Last week managed to get pyarmor registered successfully.

This week, pyarmor was reverted to (trial), 000000, non-profits. Did the entire step of pyarmor reg -g 1, the output mac was the same as the previously registered mac. Reregistering pyarmor with pyarmor-device-regfile-<*>.1.zip hits the same error.

Nothing was done over the weekend.

INFO     Python 3.10.9
INFO     Pyarmor 8.2.1 (trial), 000000, non-profits
INFO     Platform linux.x86_64
INFO     register "pyarmor-device-regfile-<*>.1.zip"
ERROR    this regfile is not for this device

Would appreciate any help to resolve this error.

```
env: air-gapped
INFO     Python 3.10.9
INFO     Pyarmor 8.2.1 (group), 00-020
INFO     Platform linux.x86_64
```

**Comment by jondy** [comment]
If the offline device hardware information is changed, please re-generate device file and re-generate offline device regfile

**Comment by alikaz3mi** [comment]
I am faciing the same issue. I followed the steps in group to work with offline unlimited dockers. Here is what I am getting in docker:

pyarmor reg pyarmor-device-regfile-6070.1.zip 
INFO     Python 3.8.10
INFO     Pyarmor 8.3.1 (trial), 000000, non-profits
INFO     Platform linux.x86_64
INFO     register "pyarmor-device-regfile-6070.1.zip"
INFO     got docker host machine id: b1601891eb532bba2b1c2af2110d0b9fa
ERROR    this regfile is not for this device

I have ran `pyarmor-auth  pyarmor-device-regfile-6070.1.zip` in a separate terminal. I had already installed pyarmor inside the docker before running it with the following command:

```
docker run --rm --gpus all --add-host=host.docker.internal:host-gateway -it cu114_trt8_torch18:latest
```

**Comment by jondy** [comment]
@alikaz3mi, please upgrade Pyarmor to v8.3.1 in all devices, then re-generate device file with id 2, and try again.

**Reply by alikaz3mi** [reply]
Thanks. But, Yesterday, just before activating my license, I have updated pyarmor in my system. The docker is in the system that I have generted device file with id 1.

**Comment by jondy** [comment]
The machine id `b1601891eb532bba2b1c2af2110d0b9fa` is generated by pyarmor 8.2.9, not the latest pyarmor.

**Reply by alikaz3mi** [reply]
I have done these stesp. But I am still getting error:


```
user@be183edfd28d:/app$ pyarmor reg pyarmor-device-regfile-6070.3.zip 
INFO     Python 3.8.10
INFO     Pyarmor 8.3.1 (trial), 000000, non-profits
INFO     Platform linux.x86_64
INFO     register "pyarmor-device-regfile-6070.3.zip"
INFO     got docker host machine id: b793aebaca007b75bc1bd7e55150c27d2
ERROR    this regfile is not for this device
```

**Comment by jondy** [comment]
> INFO     got docker host machine id: b793aebaca007b75bc1bd7e55150c27d2

The machine id should be started with `g` not `b` if it's generated by Pyarmor 8.3.1
